daemon/fqdn.go

```diff
@@ -106,17 +107,42 @@ func (d *Daemon) bootstrapFQDN(restoredEndpoints *endpointRestoreState) (err err
 		fqdn.StartDNSPoller(d.dnsPoller)
 	}
 
+	fqdn.FQDNMaxIPperHost = option.Config.ToFQDNsMaxIPsPerHost
+	// Controller to cleanup TTL expired entries from the DNS policies.
+	controller.NewManager().UpdateController("dns-garbage-collector-job", controller.ControllerParams{
+		RunInterval: 1 * time.Minute,
+		DoFunc: func() error {
+
+			namesToClean := []string{}
+			endpoints := endpointmanager.GetEndpoints()
+			for _, ep := range endpoints {
+				namesToClean = append(namesToClean, ep.DNSHistory.GC()...)
+			}
+
+			namesToClean = fqdn.KeepUniqueNames(namesToClean)
+			if len(namesToClean) == 0 {
+				return nil
+			}
+			// A second loop is needed to update the global cache from the
+			// endpoints cache
+			for i, ep := range endpoints {
+				cfg.Cache.UpdateFromCache(ep.DNSHistory, namesToClean, i == 0)
+			}
+			log.Infof("FQDN garbage collector work deleted %d name entries", len(namesToClean))
```

Comment:
this is probably a debug... I'm not sure how relevant it is without the list of names deleted

Reply:
Just a info message to be able to track this down in the future if any issue.